Made type error more clear in ACF loading

NullSoldier opened the pull request:
I was checking my logs and a user ran into this exception but I have no idea why as I'm using Python3 and pass something should be clearly a string.

Since this already throws an exception I feel like it could be slightly more helpful to the user.

leovp commented:
@NullSoldier can you change TypeError message in a couple other places too, so that they behave the same across the library?

NullSoldier commented:
Will do that shortly.

NullSoldier commented:
Modified and squashed. I could have created a TypeMismatch exception that did this type of message but I think most people expect a TypeError specifically. I think this is simple enough. What do you think?

leovp commented:
Looks great.
